src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.StateMachineMethodToClassRewriter.vb

```diff
@@ -329,28 +329,39 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Return Me.MaterializeProxy(node, Me.Proxies(Me.TopLevelMethod.MeParameter))
             End Function
 
-            Public Overrides Function VisitCatchBlock(node As BoundCatchBlock) As BoundNode
-                ' Neither Async nor Iterator function allows await/yield in Catch block,
-                ' thus, it should not capture/hoist catch block local
-                Dim rewrittenCatchLocal As LocalSymbol = Nothing
-
-                Dim origLocal As LocalSymbol = node.LocalOpt
-                If origLocal IsNot Nothing Then
-                    ' local may be either the original local of a frame reference
-                    Debug.Assert(Not Me.Proxies.ContainsKey(origLocal), "captured local should not need rewriting")
+            Private Function TryRewriteLocal(local As LocalSymbol) As LocalSymbol
+                If NeedsProxy(local) Then
+                    ' no longer a local symbol
+                    Return Nothing
+                End If
 
-                    Dim newType = VisitType(origLocal.Type)
-                    If newType = origLocal.Type Then
+                Dim newLocal As LocalSymbol = Nothing
```

Comment:
The initializer 'Nothing' is not used on any control path and looks unnecessary.